mlir/include/mlir/Dialect/GPU/IR/GPUOps.td

```diff
@@ -1250,17 +1250,17 @@ def GPU_ShuffleMode : I32EnumAttr<"ShuffleMode",
 def GPU_ShuffleModeAttr : EnumAttr<GPU_Dialect, GPU_ShuffleMode,
                                    "shuffle_mode">;
 
-def I32I64F32OrF64 : TypeConstraint<Or<[I32.predicate,
-                                        I64.predicate,
-                                        F32.predicate,
-                                        F64.predicate]>,
-                                       "i32, i64, f32 or f64">;
+def GPUShuffleTypes : TypeConstraint<Or<[
+  SignlessIntOfWidths<[8, 16, 32, 64]>.predicate,
+  FloatOfWidths<[16, 32, 64]>.predicate
+  ]>,
+  "i8, i16, i32, i64, f16, f32 or f64">;
```

Comment:
Why not allow any SignlessInt or Float and filter when lowering?

Reply:
Also that seems a bit more disruptive and I believe this should be enough for our usecase.

Reply:
Yeah, I guess this should pretty much work fine